R/utilities-dimensions.R

```diff
@@ -0,0 +1,162 @@
+#' of both dimensions.
+#' If the values cannot be transformed between the dimensions, an error is thrown.
+#' @export
+dimensionsConversionFactor <- function(dimension1, dimension2, mw = NULL) {
+  # No conversion if the dimensions are equall
+  validateIsString(c(dimension1, dimension2))
+  if (dimension1 == dimension2) {
+    return(1)
+  }
+
+  # Dimensionless <-> Fraction
+  if (dimension1 == Dimensions$Dimensionless) {
+    if (dimension2 == Dimensions$Fraction) {
+      return(1)
+    }
+  }
+  if (dimension1 == Dimensions$Fraction) {
+    if (dimension2 == Dimensions$Dimensionless) {
+      return(1)
+    }
+  }
+
+  # Concentration (molar) <-> Concentration (mass)
+  if (dimension1 == Dimensions$`Concentration (molar)`) {
+    if (dimension2 == Dimensions$`Concentration (mass)`) {
+      if (is.null(mw)) {
+        stop(messages$errorCannotConvertDimensionsNoMW(dimension1, dimension2))
+      }
+      return(1 * 1e-6 * mw * 1e-3)
+    }
+  }
+  if (dimension1 == Dimensions$`Concentration (mass)`) {
+    if (dimension2 == Dimensions$`Concentration (molar)`) {
+      if (is.null(mw)) {
+        stop(messages$errorCannotConvertDimensionsNoMW(dimension1, dimension2))
+      }
+      return(1 * 1e3 / mw * 1e6)
+    }
+  }
+
+  # Amount (molar) <-> Mass
+  if (dimension1 == Dimensions$Amount) {
+    if (dimension2 == Dimensions$Mass) {
+      if (is.null(mw)) {
+        stop(messages$errorCannotConvertDimensionsNoMW(dimension1, dimension2))
+      }
+      return(1 * 1e-6 * mw * 1e-3)
+    }
+  }
+  if (dimension1 == Dimensions$Mass) {
+    if (dimension2 == Dimensions$Amount) {
+      if (is.null(mw)) {
+        stop(messages$errorCannotConvertDimensionsNoMW(dimension1, dimension2))
+      }
+      return(1 * 1e3 / mw * 1e6)
+    }
+  }
+
+  stop(messages$errorCannotConvertDimensions(dimension1, dimension2))
+}
+
+#' Get conversion factor from one unit to another
+#'
+#' @param fromUnit String name of the unit to convert from
+#' @param toUnit String name of the unit to convert to
+#' @param dimension Dimension of the units
+#'
+#' @return Numerical value. A value in \code{fromUnit} multiplied  by the factor
+#' corresponds to the value in \code{toUnit}
+#' @export
+getUnitConversionFactor <- function(fromUnit, toUnit, dimension) {
+  validateDimension(dimension)
+  validateUnit(fromUnit, dimension)
+  validateUnit(toUnit, dimension)
+
+  dimensionTask <- getDimensionTask()
+  dimension <- rClr::clrCall(dimensionTask, "DimensionByName", enc2utf8(dimension))
+  fromUnitObj <- rClr::clrCall(dimension, "UnitOrDefault", enc2utf8(fromUnit))
+  toUnitObj <- rClr::clrCall(dimension, "UnitOrDefault", enc2utf8(toUnit))
+  convFac <- rClr::clrCall(dimension, "UnitValueToBaseUnitValue", fromUnitObj, 1)
+  convFac <- convFac * rClr::clrCall(dimension, "BaseUnitValueToUnitValue", toUnitObj, 1)
+  return(convFac)
+}
+
+#' #' Supported dimensions.
+#' #'
+#' #' @include enum.R
+#' #'
+#' #' @export
+#' Dimensions <- enum(allAvailableDimensions())

```

Comment:
This is a problem here. It is being loaded directly before initialization. This works when you load ospsuite.core first but does not work in ospsuite.core